### DISCUSSION TITLE
Olimex ESP32 POE mit nRF & CMT

### DISCUSSION THREAD

**Discussion by Rookwello** [tbnobody/OpenDTU#1041] [Q&A]
Kann man an einem Olimex ESP32 POE nRF & CMT gleichzeitig nutzen?
Mit nRF läuft es problemlos. Sobald ich das CMT aktiviere, geht kein Ethernet mehr.
Ich hab's mit folgendem Mapping probiert:
```
"name": "Olimex ESP32-POE with CMT",
        "nrf24": {
            "miso": 15,
            "mosi": 2,
            "clk": 14,
            "irq": 13,
            "en": 16,
            "cs": 5
        },
        "cmt": {
            "clk": 36,
            "cs": 33,
            "fcs": 32,
            "gpio2": -1,
            "gpio3": -1,
            "sdio": 35
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        }
```

**Comment by taubenhorst** [comment]
Does anyone have a working device profile for an olimex poe board with cmt and nrf? Many thanks in advance.

**Reply by vaterlangen** [reply]
I also had problems with the Olimex ESP32 POE board, when CMT was connected to GPIO0 and GPIO1. Board did not start network connection (but was PoE powered) on POR - after Soft-Reset via WLAN the PHY came up. I assume, it has someting to do with the Pins beeing used somehow other during POR and moved to other GPIOs without further investigation. Removing or disabling the CMT fixed the issue.
Maybe you have a similar problem - when I look at your GPIO mapping, I can see clk and sdio beeing mapped to GPI pins (I'm not familiar with ESP32, but I would expect these pins are input only). Acording to the E49-900M20S-Usermanual, CLK is an input on the CMT and SDIO is bidirectional. Maybe moveing these pins to other GPIOs (e.g. 3 and 4) may help?

**Comment by turrican944** [comment]
Moin
Ich habe es jetzt bei mir am laufen, zumindest ein schneller Test auf dem Steckbrett hat funktioniert.
![PXL_20240418_060759054](https://github.com/tbnobody/OpenDTU/assets/70096465/486e21b8-6710-4455-b86e-8f4f617c81bd)
[Device Profil D1 ESP32.json](https://github.com/tbnobody/OpenDTU/files/15021067/Device.Profil.D1.ESP32.json)

 NRF + CMT + Display funktioniert
 
        "name": "Olimex ESP32-POE CMT2003a + NRF",
        "cmt": {
            "clk": 15,
            "cs": 16,
            "fcs": 13,
            "sdio": 2
        },
	"nrf24": {
            "miso": 36,
            "mosi": 32,
            "clk": 4,
            "irq": 35,
            "en": 14,
            "cs": 5
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        },
        "led": {
            "led0": -1,
            "led1": -1
	},
        "display": {
            "type": 3,
            "data": 33,
            "clk": 0
        }

Ich habe für das NRF IRQ und MISO bewusst die GPI35 und 36 ausgwählt da diese Pins am Olimex nur Eingänge sind, sonst hätte ich keine GPIOs mehr für das Display gehabt. GPIO0 und 1 für Display oder NRF geht nicht, dann fährt der Olimex nicht hoch.
In der Deviceprofil sind aber diverse Profile drin, das richtige auswählen.

GPIO1 und GPIO3 sind die RS232 des Olimex die auch aufs USB gehen, da kann man aber auch die LEDs drauf legen aber dann kommen keine debug ausgaben mehr per USB. Deshalb habe ich den SCL im Display auf GPIO0 geändert.

@Rookwello Deine Pinconfig kann auch nicht funktionieren, denn du hast
  "clk": 36,  "sdio": 35 für das CMT genommen, die Pins 36 und 35 sind aber nur Eingänge  am ESP32. Da soll bei dir aber Clock rauskommen und Data I/O deshalb geht es nicht.
